### PR TITLE
doc: module and submodule readme

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description

Global README and submodule README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,13 @@
-# {Module-Name} Terraform Module
+# Push Based Log Export Terraform Module
 
-Description of the module
+This Terraform module configures push-based log export in an Atlas project. Consists of the following submodule:
```
Comment (corryroot) [inline README.md]:
```suggestion
This Terraform module configures push-based log export in an Atlas project and consists of the following submodule:
```

File: modules/s3_bucket/README.md [hunk 2]
```diff
@@ -1,16 +1,54 @@
...
+| [aws_iam_role](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role) | resource |
+| [aws_iam_role_policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy) | resource |
+| [aws_s3_bucket](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/s3_bucket) | data source | resource |
+
+Refer to the [MongoDB Atlas](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs) and [AWS](https://registry.terraform.io/providers/hashicorp/aws/latest/docs) Terraform providers documentations if you want more information.
+
+## Considerations
 
-## Considerations 
+If you set the input variable `force_destroy` to true, if you dereference the module the AWS S3 Bucket will also be destroyed together with the rest of the resources, even if it has content and files inside.
```
Comment (corryroot) [inline modules/s3_bucket/README.md]:
```suggestion
When you set the input variable `force_destroy` to true, if you dereference the module, the AWS S3 Bucket will also be destroyed together with the rest of the resources, even if it has content and files inside.
```

**Review by corryroot — APPROVED** [review]
LGTM! I commented with some copy nits.

File: modules/s3_bucket/README.md [hunk 3]
```diff
@@ -1,16 +1,54 @@
...
+  project_id = "66a26b4c85718b1be4ff37cb"
+  bucket_name = "my-bucket"
+  create_bucket = true
+  iam_role_name = "push-log-role"
+  iam_role_policy_name = "push-log-policy"
+  force_destroy = false
+}
+```
+
+The [examples](https://github.com/terraform-mongodbatlas-modules/terraform-mongodbatlas-push-based-log-export/tree/main/examples) folder contains detailed examples that show how to use this submodule.
```
Comment (EspenAlbert) [inline modules/s3_bucket/README.md]: Potentially use a list to link to each example with its "one-line" description

File: modules/s3_bucket/README.md [hunk 4]
```diff
@@ -1,16 +1,54 @@
...
+- An AWS S3 Bucket.
+- An AWS IAM Role Policy.
+- A MongoDB Atlas Cloud Provider Access Authorization.
+- Enables Push-Based Log Export in an Atlas project.
+
+You can find detailed information of the submodule's input and output variables in the [Terraform Public Registry]()
 
 ## Usage 
 
+```hcl
```
Comment (EspenAlbert) [inline modules/s3_bucket/README.md]: [nit] We are trying to use ```terraform in readme code snippets

**Review by EspenAlbert — APPROVED** [review]
Only minor comments

File: modules/s3_bucket/README.md [hunk 5]
```diff
@@ -1,16 +1,58 @@
...
+| [aws_iam_role](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role) | resource |
+| [aws_iam_role_policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy) | resource |
+| [aws_s3_bucket](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/s3_bucket) | data source | resource |
+
+Refer to the [MongoDB Atlas](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs) and [AWS](https://registry.terraform.io/providers/hashicorp/aws/latest/docs) Terraform providers documentations if you want more information.
+
+## Considerations
 
-## Considerations 
+When you set the input variable `force_destroy` to true, if you dereference the module, the AWS S3 Bucket will also be destroyed together with the rest of the resources, even if it has content and files inside.
```
Comment (AgustinBettati) [inline modules/s3_bucket/README.md]: The term `dereference` sounds unclear to me, would `remove the module` be more explicit?

File: modules/s3_bucket/README.md [hunk 6]
```diff
@@ -1,16 +1,58 @@
...
+# S3 Bucket Terraform Submodule
 
-Description of the module
+This Terraform submodule configures [push-based log export](https://www.mongodb.com/docs/atlas/push-logs/) to an AWS S3 Bucket in an Atlas project. 
+
+It creates the following resources:
+
+- A MongoDB Atlas Cloud Provider Access Setup.
+- An AWS IAM Role.
+- An AWS S3 Bucket.
```
Comment (AgustinBettati) [inline modules/s3_bucket/README.md]: Should we clarify S3 is created depending on input parameters used?

File: README.md [hunk 7]
```diff
@@ -1,16 +1,13 @@
-# {Module-Name} Terraform Module
+# Push Based Log Export Terraform Module
 
-Description of the module
+This Terraform module configures push-based log export in an Atlas project and consists of the following submodule:
 
-## Usage 
+- [s3 bucket](https://github.com/terraform-mongodbatlas-modules/terraform-mongodbatlas-push-based-log-export/tree/main/modules/s3_bucket)
```
Comment (AgustinBettati) [inline README.md]: nit: I see we define the s3 sub module directory with snake case `s3_bucket`, is there a reason for this? Was seeing that aws modules use dash case ([reference](https://github.com/terraform-aws-modules/terraform-aws-iam/tree/master/modules)), not sure if there is reference of other modules that are using snake case.

**Review by AgustinBettati — APPROVED** [review]
LGTM, some small comments

File: README.md (hunk 7 above)
Comment (rubenVB01) [reply README.md]: was taking as reference the google one: https://github.com/terraform-google-modules/terraform-google-sql-db/tree/master/modules but if it's more common to use dash case I have no concerns changing it

Comment (AgustinBettati) [reply README.md]: 👍 I see. It appears azure also uses snake case https://registry.terraform.io/modules/aztfmod/caf/azurerm/latest, we can stick with this and make sure to be consistent within our modules.

**Pull request merged**